### PR TITLE
PreviousMixingに重み入れるの忘れてた

### PR DESCRIPTION
**Pull request by myxyy**
# 概要
重みパラメータが無かった  
ついでにclear_hiddenまわりの挙動も修正

## Submit前の確認項目

- [x] タイトルは一目でわかるようにし、説明文は PR を簡潔に説明するようにしましたか?
- [x] あなたの PR が、異なる変更を束ねたものではなく、ひとつのことだけを行うものであることを確認しましたか?
- [x] この PR で導入されたすべての変更点をリストアップしましたか?
- [x] PR を `pytest`または `make test`コマンドでローカルにテストしましたか?
- [x] `pre-commit run -a` または`make format`コマンドで `pre-commit` フックを実行しましたか?

### CODE REVIEW

**Comment by Geson-anko** [comment]
@myxyy 現在のmainからブランチを切るか、rebaseしてください。

**Comment by myxyy** [comment]
@Geson-anko rebaseしました

File: src/models/components/previous_mixing.py [hunk 1]
```diff
@@ -8,16 +8,17 @@ def __init__(self, dim: int):
...
-            self.x_last = torch.zeros(x.shape[1:], dtype=x.dtype, device=x.device)
+            self.clear_hidden()
         x_shift = x.roll(shifts=1, dims=0)
         x_shift[0] = self.x_last
         self.x_last = x[-1]
-        return x * self.mix_factor + x_shift * (1 - self.mix_factor)
+        return self.W(x * self.mix_factor + x_shift * (1 - self.mix_factor))
 
     def clear_hidden(self):
-        self.x_last = None
```
Comment (Geson-anko) [inline src/models/components/previous_mixing.py:23]: おそらく、`self.W = nn.Linear(dim, dim, bias=False)`と`return self.W(x * self.mix_factor + x_shift * (1 - self.mix_factor))`のところだけ変更したと思いますが、それ以外の箇所も変更されています。rebaseで直ると思ったんですが、mainブランチから切りなおして修正することをお勧めします。

**Review by Geson-anko — APPROVED** [review]
@myxyy rebaseで 直らなかったので新しくmainからブランチを切って修正してみてください。

**Review by Geson-anko — CHANGES_REQUESTED** [review]
@myxyy ミスってapproveしちゃいました。修正お願いします

File: src/models/components/previous_mixing.py (hunk 1 above)
Comment (myxyy) [reply src/models/components/previous_mixing.py:23]: すいません、概要に書いておくべきでしたがついでにclean_hidden周りの挙動も直しています  
あまり行儀のよいcommitではなかったですね

**Comment by Geson-anko** [comment]
@myxyy やっぱり重みを適用するのは`ChannelMixing`側でやることを提案します。
コンポーネントの役割を独立化するために、`PreviousMixing`というコンポーネントに持たせる機能は前トークンと混ぜるという機能だけに絞るとよいでしょう。また、今回は`ChannelMixing`内部のチャネル数を4倍にするといった処理を行いませんが、一般的にそのような内部チャネルのサイズを変更するといったときに、PreviousMixingにinputとoutputのサイズを指定を引数に加えなければならなくなり煩雑化します。

したがって、今回のPull Requestをマージせず、ChannelMixingを変更することによって今回の修正を適用することを提案します。
